exonum-java-binding/core/rust/integration_tests/tests/runtime.rs

```diff
@@ -0,0 +1,659 @@
+
+    let tx = create_transaction(instance_id, fake_service::SET_ENTRY_TX, tx_args);
+    let block = test_kit.create_block_with_transaction(tx);
+
+    assert_eq!(block.transactions.len(), 1);
+    assert!(block.transactions.get(0).unwrap().status().is_ok());
+
+    test_kit.create_block();
+
+    let snapshot = test_kit.snapshot();
+    let test_entry = create_tx_test_entry(&*snapshot, fake_service::TEST_SERVICE_NAME);
+    let value = test_entry
+        .get()
+        .expect("Failed to find the entry created in the test service");
+
+    assert_eq!(tx_args_hex, value);
+
+    test_kit.stop();
+}
+
+#[test]
+fn submit_non_convertible_tx() {
+    let tx_args = &[1_u8, 2_u8, 3_u8, 4_u8];
+
+    let (runtime, instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+    test_kit.create_block();
+
+    let tx = create_transaction(instance_id, fake_service::THROW_SOE_TX, tx_args);
+    let block = test_kit.create_block_with_transaction(tx);
+
+    assert_eq!(block.transactions.len(), 1);
+    let status = block.transactions.get(0).unwrap().status();
+    assert!(status.is_err());
+    assert!(status
+        .unwrap_err()
+        .to_string()
+        .contains("java.lang.StackOverflowError;"));
+
+    test_kit.stop();
+}
+
+#[test]
+fn submit_service_error_on_exec_tx() {
+    let tx_args = &[255_u8, 2_u8, 3_u8, 4_u8];
+    let tx_error_code = tx_args[0];
+
+    let (runtime, instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+
+    test_kit.create_block();
+
+    let tx = create_transaction(instance_id, fake_service::SRVC_ERR_ON_EXEC_TX, tx_args);
+    let block = test_kit.create_block_with_transaction(tx);
+
+    assert_eq!(block.transactions.len(), 1);
+    let status = block.transactions.get(0).unwrap().status();
+    assert!(status.is_err());
+    assert_service_error_with_code(status.unwrap_err(), tx_error_code);
+
+    test_kit.stop();
+}
+
+#[test]
+fn submit_failing_on_exec_tx() {
+    let tx_args = &[1_u8, 2_u8, 3_u8, 4_u8];
+
+    let (runtime, instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+
+    test_kit.create_block();
+
+    let tx = create_transaction(instance_id, fake_service::FAIL_ON_EXEC_TX, tx_args);
+    let block = test_kit.create_block_with_transaction(tx);
+
+    assert_eq!(block.transactions.len(), 1);
+    let status = block.transactions.get(0).unwrap().status();
+    assert!(status.is_err());
+
+    // Strange Core behavior, stored result is ExecutionError { err.kind == Service, err.code = 1 }
```

Comment:
Was Core team contacted? Is it confirmed to be Core bug, not our implementation flaw?